tests/test_cusparseSolver.cpp

```diff
@@ -129,24 +117,70 @@ testCusparseSolver(const boost::property_tree::ptree& prm, Matrix<bz>& matrix, V
     }
 }
 
+template <int bz>
+Dune::BlockVector<Dune::FieldVector<double, bz>>
+testCusparseSolver(std::unique_ptr<Opm::BdaBridge<Matrix<bz>, Vector<bz>, bz> >& bridge, Matrix<bz>& matrix, Vector<bz>& rhs)
```

Comment:
Just a general comment: Using `std::unique_ptr<T>&` as a parameter type **typically** announces that the function wants to participate in the lifetime management of the contained object and is going to change the `unique_ptr<>` object itself, not that we just want to use the underlying object.

Unless I'm missing something, this function does not appear to alter the actual `unique_ptr`, so in this case I'd probably implement the function as follows
```C++
template <int bz>
Dune::BlockVector<Dune::FieldVector<double, bz>>
testCusparseSolver(Opm::BdaBridge<...>& bridge, const Matrix<bz>& matrix, const Vector<bz>& rhs)
{
    // ...
    bridge.solve_system(...);
    bridge.get_result(x);

    return x;
}
```
and call it as
```
testCusparseSolver(*bridge, matrix, rhs);
```
instead.

Reply:
Good point, I'll change it